exetera/core/dataframe.py

```diff
@@ -434,6 +493,15 @@ def apply_index(self, index_to_apply, ddf=None):
         """
         Apply the index to all the fields in this dataframe, return a dataframe with indexed fields.
 
+        Example::
+            # apply index inplace
+            index = np.array([4, 3, 2, 1, 0])
+            df.apply_index(index)
+
+            # apply index and store new result to designated dataframe
+            df.apply_index(index, ddf=df2)
+
+
         :param index_to_apply: the index to be applied to the fields, an ndarray of integers
```

Comment:
For example here, as a user, I would want to see the output of carrying out this action. In general, if you are showing code, you should show the result of the code.

Reply:
We would want to create a dataframe with the appropriate data but we don't have to do that part correctly since it's explained elsewhere. Having an example like this in the docstring would work:

```python
df = .... # contains ["a", "b", "c", "d", "e", "f"]
index = np.array([4, 3, 2, 0, 1])
df1 = df.apply_index(index)
print(df1) # prints ["e", "d", "c", "a", "b"]
```